ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackend.scala

```diff
@@ -809,43 +825,106 @@ private[backend] trait CommonStorageBackend[DB_BATCH] extends StorageBackend[DB_
       .asVectorOf(contractStateRowParser)(connection)
 
   // Events
-
-  def pruneEvents(pruneUpToInclusive: Offset)(connection: Connection): Unit = {
+  def pruneEvents(pruneUpToInclusive: Offset, pruneAllDivulgedContracts: Boolean)(
+      connection: Connection
+  ): Unit = {
     import com.daml.platform.store.Conversions.OffsetToStatement
-    List(
+    val conditionalImmediateDivulgencePruning =
+      pruneImmediateDivulgence(pruneUpToInclusive, pruneAllDivulgedContracts)
```

Comment:
I'd suggest to inline the SQL code here once the query has shrunk in size. This way the different statement can easily be compared.

Reply:
Updated according to the discussion had offline.